frame/transaction-payment/src/lib.rs

```diff
@@ -515,6 +515,7 @@ impl<T> Convert<Weight, BalanceOf<T>> for Pallet<T> where
 /// Require the transactor pay for themselves and maybe include a tip to gain additional priority
 /// in the queue.
 #[derive(Encode, Decode, Clone, Eq, PartialEq, TypeInfo)]
+#[scale_info(skip_type_params(T))]
```

Comment:
This should follow what other crates are doing and should be `custom_bounds` or similar

Reply:
skip type params is not exactly about bounds.
By default, the type info of a type contains the type info of its generic type. No matter how the generic type is used in the fields of the type.
This attribute allow to not put the generic parameter `T` into the type info of `ChargeTransactionPayment`. 

honestly i don't really know why we put the generic parameter into the type info. So I don't know if it is ok to skip them in general. But here `T` being the runtime, knowing that the param is the type `Runtime` doesn't seem useful.

But I'm also ok to bound TypeInfo to the trait `frame_system::Config`.

Reply:
Ahh okay. Ty for the explanation

Reply:
> This should follow what other crates are doing and should be custom_bounds or similar

We have a `bounds` attribute too, see the PR description here for an explanation: https://github.com/paritytech/scale-info/pull/96.

> honestly i don't really know why we put the generic parameter into the type info. So I don't know if it is ok to skip them in general. But here T being the runtime, knowing that the param is the type Runtime doesn't seem useful.

That is a good question, it was a precursor to representing generics in the type information, which we ultimately gave up on https://github.com/paritytech/scale-info/pull/12.

However the current downstream use case I have for it is to correctly "demonomorphize" :eyes: the concrete types into generic ones when generating Rust types e.g. https://github.com/ascjones/chameleon/blob/master/core/src/generate_types.rs#L95. That said, this could probably be done by parsing the full type paths and matching with generic type parameter names  instead of concrete types - though that would need some careful thought as it opens up the generics can of worms again.